src/information_schema.rs

```diff
@@ -0,0 +1,795 @@
+
+        RecordBatch::try_new(
+            self.schema.clone(),
+            vec![schema_names, table_names, file_paths, file_sizes, record_counts, has_delete_file],
+        )
+        .map_err(|e| datafusion::error::DataFusionError::ArrowError(Box::new(e), None))
+    }
+}
+
+#[async_trait::async_trait]
+impl TableProvider for FilesTable {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.schema.clone()
+    }
+
+    fn table_type(&self) -> TableType {
+        TableType::Base
+    }
+
+    async fn scan(
+        &self,
+        state: &dyn Session,
+        projection: Option<&Vec<usize>>,
+        filters: &[datafusion::prelude::Expr],
+        limit: Option<usize>,
+    ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
+        // Query catalog database live
+        let batch = self.query_files()?;
+
+        // Use MemTable for execution
+        let mem_table = MemTable::try_new(self.schema.clone(), vec![vec![batch]])?;
+        mem_table.scan(state, projection, filters, limit).await
+    }
+}
+
+/// Schema provider for information_schema
+///
+/// Provides live metadata tables that query the catalog database on every access.
+/// No upfront data loading - all queries execute fresh against the metadata provider.
+#[derive(Debug)]
+pub struct InformationSchemaProvider {
+    provider: Arc<dyn MetadataProvider>,
+}
+
+impl InformationSchemaProvider {
+    pub fn new(provider: Arc<dyn MetadataProvider>) -> Self {
+        Self {
+            provider,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl SchemaProvider for InformationSchemaProvider {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn table_names(&self) -> Vec<String> {
+        vec![
+            "snapshots".to_string(),
+            "schemata".to_string(),
+            "tables".to_string(),
+            "table_info".to_string(),
+            "columns".to_string(),
+            "files".to_string(),
+        ]
+    }
+
+    async fn table(&self, name: &str) -> DataFusionResult<Option<Arc<dyn TableProvider>>> {
+        // Create table provider on-demand - queries will be live
+        let provider: Option<Arc<dyn TableProvider>> = match name {
+            "snapshots" => Some(Arc::new(SnapshotsTable::new(self.provider.clone()))),
+            "schemata" => Some(Arc::new(SchemataTable::new(self.provider.clone()))),
+            "tables" => Some(Arc::new(TablesTable::new(self.provider.clone()))),
+            "table_info" => Some(Arc::new(TableInfoTable::new(self.provider.clone()))),
+            "columns" => Some(Arc::new(ColumnsTable::new(self.provider.clone()))),
+            "files" => Some(Arc::new(FilesTable::new(self.provider.clone()))),
+            _ => None,
+        };
+        Ok(provider)
+    }
+
+    fn table_exist(&self, name: &str) -> bool {
+        matches!(
+            name,
+            "snapshots" | "schemata" | "tables" | "table_info" | "columns" | "files"
+        )
```

Comment:
```suggestion
        self.table_names().iter().any(|t| t == name)
```
probably makes sense to make table_names the source of truth for the tables.